packages/core/src/shaderlib/particle/rotation_over_lifetime_module.glsl

```diff
@@ -41,69 +41,53 @@ float computeParticleRotationFloat(in float rotation, in float age, in float nor
 }
 
 
-#if defined(RENDERER_MODE_MESH) && (defined(ROTATION_OVER_LIFETIME) || defined(ROTATION_OVER_LIFETIME_SEPARATE))
-vec3 computeParticleRotationVec3(in vec3 rotation,
-    in float age,
-    in float normalizedAge) {
-#ifdef ROTATION_OVER_LIFETIME
-    #ifdef ROTATION_OVER_LIFETIME_CONSTANT
-        float ageRot = u_ROLAngularVelocityConst * age;
-        rotation += ageRot;
-    #endif
-    #ifdef ROTATION_OVER_LIFETIME_CURVE
-        rotation += getTotalValueFromGradientFloat(u_ROLAngularVelocityGradient, normalizedAge);
-    #endif
-    #ifdef ROTATION_OVER_LIFETIME_RANDOM_CONSTANTS
-        float ageRot = mix(u_ROLAngularVelocityConst, u_ROLAngularVelocityConstMax, a_Random0.w) * age;
-        rotation += ageRot;
-    #endif
-    #ifdef ROTATION_OVER_LIFETIME_RANDOM_CURVES
-        rotation += mix(
-        getTotalValueFromGradientFloat(u_ROLAngularVelocityGradient, normalizedAge),
-        getTotalValueFromGradientFloat(u_ROLAngularVelocityGradientMax,
-            normalizedAge),
-        a_Random0.w);
-    #endif
-#endif
+#if defined(RENDERER_MODE_MESH) && (defined(RENDERER_ROL_CONSTANT_MODE) || defined(RENDERER_ROL_CURVE_MODE))
+vec3 computeParticleRotationVec3(in vec3 rotation, in float age, in float normalizedAge) {
+    #ifdef RENDERER_ROL_IS_SEPARATE
+        #ifdef RENDERER_ROL_CONSTANT_MODE
+            #ifdef RENDERER_ROL_IS_RANDOM_TWO
+                vec3 ageRot = mix(renderer_ROLMinConst, renderer_ROLMaxConst, a_Random0.w) * age;
+            #else
+                vec3 ageRot = renderer_ROLMaxConst * age;
+            #endif
+            rotation += ageRot;
+        #endif
+        #ifdef RENDERER_ROL_CURVE_MODE
+            #ifdef RENDERER_ROL_IS_RANDOM_TWO
+                rotation += vec3(
+                mix(getTotalValueFromGradientFloat(renderer_ROLMinCurveX, normalizedAge),
```

Comment:
Curve mode is not supported now, you haven't implemented methods like `renderer_ROLMaxCurveX` and `evaluateParticleCurveCumulative`.

Reply:
To be resolved later